UI/ChatGPT/ChatGPT/ChatGPT/Presentation/Message.cs

```diff
@@ -9,4 +11,13 @@ public Message(string userMessage)
 		: this(Guid.NewGuid(), Source.User, Status.Value, userMessage)
 	{
 	}
+
+	public Message Update(ChatResponse response)
```

Comment:
I'm not sure I'm a fan of using Update here as it implies changing current instance. Perhaps "With" to be consistent with how records are updated
eg msg.With(response)
cc @francoistanguay thoughts?

Reply:
Fait. Maybe "With" as a With(reponse) : Message...

We'll see those patterns evolve... in ToDo... we had Todo.ToggleIsCompleted():ToDo as itwas modifying only one field and not accepting a param...

I dont think a method named .Response(reponse) would make much sense but maybe?

It's just a sample but I can see why Update has more mutable meaning and could be congusing.

Reply:
Maybe `message.WithResponse(response)` would be more concise?

Reply:
Sure. That's fine. As it was the only "With" or that we could even have overloads... I was fine with .With simply... But if you prefer yours, go for it.